editor/store/effects.js

```diff
@@ -498,11 +501,18 @@ export default {
 		}, {} );
 		store.dispatch( setMetaBoxSavedData( dataPerLocation ) );
 
+		// Additional data needed for backwards compatibility.
+		// If we do not provide this data the post will be overriden with the default values.
+		const additionData = []
+			.concat( post[ 'comment_status' ] ? [ `comment_status=${ post[ 'comment_status' ] }` ] : [] )
+			.concat( post[ 'ping_status' ] ? [ `ping_status=${ post[ 'ping_status' ] }` ] : [] );
+
 		// To save the metaboxes, we serialize each one of the location forms and combine them
 		// We also add the "common" hidden fields from the base .metabox-base-form
-		const formData = values( dataPerLocation ).concat(
-			jQuery( '.metabox-base-form' ).serialize()
-		).join( '&' );
+		const formData = values( dataPerLocation )
+			.concat( jQuery( '.metabox-base-form' ).serialize() )
+			.concat( additionData )
+			.join( '&' );
```

Comment:
Aside: Would it be easier to construct this with [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData) instead?

Unfortunately not possible to pass an object to its constructor, needs to iterate with [`FormData#append`](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append).†

But it can be passed directly into `fetch`'s `body` property.

Above `jQuery.serialize` would need to be revised to [`jQuery.serializeArray`](https://api.jquery.com/serializeArray/).

I think `FormData` manages encoding too, unlike the current code here where we concatenate `post[ 'comment_status' ]` into the form without first `encodeURIComponent` (not likely a real-world issue).

_† There are helpers like [`object-to-formdata`](https://www.npmjs.com/package/object-to-formdata)_

Reply:
I’m afraid FormData doesn’t work properly with our codebase. We polyfill fetch with very limited support for FormData. At least I had a few issues with a few weeks back.

Reply:
Please note the gaps for Edge and Safari: https://developer.mozilla.org/en-US/docs/Web/API/FormData.

Reply:
I tried something similar the problem is that `serializeArray` doesn't return a nested array when we have array parameters instead it returns something like `{ "fields[foo]": "bar" }` which is not the correct object we want.